views/default/page/components/table.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+/**
+ * View a table of items
+ *
+ * @uses $vars['cell_views']   Array of view names used to render each column. E.g. the default is
+ *                             ['page/components/table/cell/default'], which specifies one column
+ *                             containing the item.
+ *
+ * @uses $vars['headings']     Array of HTML headings (each will be wrapped by TH elements)
+ *
+ * @uses $vars['items']        Array of ElggEntity, ElggAnnotation or ElggRiverItem objects
+ * @uses $vars['offset']       Index of the first list item in complete list
+ * @uses $vars['limit']        Number of items per page. Only used as input to pagination.
+ * @uses $vars['count']        Number of items in the complete list
+ * @uses $vars['base_url']     Base URL of list (optional)
+ * @uses $vars['url_fragment'] URL fragment to add to links if not present in base_url (optional)
+ * @uses $vars['pagination']   Show pagination? (default: true)
+ * @uses $vars['position']     Position of the pagination: before, after, or both
+ * @uses $vars['full_view']    Show the full view of the items (default: false)
+ * @uses $vars['list_class']   Additional CSS class for the <table> element
+ * @uses $vars['item_class']   Additional CSS class for the <td> elements
+ * @uses $vars['item_view']    Alternative view to render list items
+ * @uses $vars['no_results']   Message to display if no results (string|Closure)
+ */
+$items = $vars['items'];
+$count = elgg_extract('count', $vars);
+$pagination = elgg_extract('pagination', $vars, true);
+$position = elgg_extract('position', $vars, 'after');
+$no_results = elgg_extract('no_results', $vars, '');
+$cell_views = elgg_extract('cell_views', $vars, ['page/components/table/cell/default']);
+$headings = elgg_extract('headings', $vars, '');
+
+if (empty($cell_views) || !is_array($cell_views)) {
+	elgg_log('$vars["cell_views"] must be an array of view names', 'ERROR');
+	return;
+}
+
+if (is_array($headings)) {
+	if (count($headings) !== count($cell_views)) {
+		// complain but carry on rendering
+		elgg_log('$vars["headings"] size must match $vars["cell_views"]', 'ERROR');
+		$headings = '';
+	} else {
+		$heading_renderer = function ($heading) {
+			return "<th>$heading</th>";
+		};
+		$headings = array_map($heading_renderer, $headings);
+		$headings = implode('', $headings);
+		$headings = "<thead><tr>$headings</tr></thead>";
+	}
+}
+
+if (!$items && $no_results) {
+	if ($no_results instanceof Closure) {
+		echo $no_results();
+		return;
+	}
+	echo "<p class='elgg-no-results'>$no_results</p>";
+	return;
+}
+
+if (!is_array($items) || count($items) == 0) {
+	return;
+}
+
+$table_classes = ['elgg-list', 'elgg-table'];
+if (isset($vars['list_class'])) {
+	$table_classes[] = $vars['list_class'];
+}
+
+$cell_classes = ['elgg-item'];
+if (isset($vars['item_class'])) {
+	$cell_classes[] = $vars['item_class'];
```

Comment:
I would think `item_class` should map to `<tr>` class. Cells can get their class from heading.